rotkehlchen/api/rest.py

```diff
@@ -4106,52 +3315,18 @@ def _get_avalanche_transactions(
         msg = ''
         return {'result': result, 'message': msg, 'status_code': HTTPStatus.OK}
 
-    def get_avalanche_transactions(
-            self,
-            async_query: bool,
-            address: ChecksumEvmAddress,
-            from_timestamp: Timestamp,
-            to_timestamp: Timestamp,
-    ) -> Response:
-        if async_query is True:
-            return self._query_async(
-                command=self._get_avalanche_transactions,
-                address=address,
-                from_timestamp=from_timestamp,
-                to_timestamp=to_timestamp,
-            )
-
-        response = self._get_avalanche_transactions(
-            address=address,
-            from_timestamp=from_timestamp,
-            to_timestamp=to_timestamp,
-        )
-
-        result = response['result']
-        if len(result) == 0:
-            return api_response(
-                wrap_in_fail_result('Not found.'),
-                status_code=HTTPStatus.NOT_FOUND,
-            )
```

Comment:
This is missing from the new version if I'm correct

Reply:
Why do you think that it is missing? Isn't it handled here https://github.com/rotki/rotki/blob/ae7b18368fc4cbc0b45a94808d102287d6371b52/rotkehlchen/api/rest.py#L3301 ?

Reply:
That checks if response is None but not if `len(response) == 0`. In the removed code `[]` also returns status code NOT FOUND

Reply:
Previously async wrapper function `get_avalanche_transactions` was reading the result of the function with real logic `_get_avalanche_transactions` and doing some modifications of the status codes on top of it (e.g. getting the `result` and forcing status code to `NOT_FOUND` if length of the result is 0).

Now, since the status codes and messages of sync and async paths of execution became the same, there is now more consistency. Which is good.

Considering the fact that avax transactions endpoint currently is not used by the frontend, this change doesn't break anything, but just gives more consistency for the future, so I think it is a good change.

Reply:
No this is not sufficient, yabir is absolutely right? What do you think? That hiding bugs for the future is okay so long as things are "consistent"? Absolutely not. Whether or not something is used plays no role at all, because things change and when they do this is a serious bug which will bite us.

Fortunately it's very easy to solve. The only reason an empty list can be returned there is due to the implementation of what used to be `_get_avalanche_transactions` and what is now `get_avalanche_transactions` in this same file. Don't you see it?

Can't easily link since you are editing it in this PR but look here:

```python
        except RemoteError as e:
            return {
                'result': [],
                'message': f'{str(e)}',
                'status_code': HTTPStatus.BAD_GATEWAY,
            }
        if response is None:
            return {
                'result': [],
                'message': 'Not found.',
                'status_code': HTTPStatus.NOT_FOUND,
            }
```

It returns empty list in both error cases, which is an error in itself :). You know what to do.